### PR TITLE
bugfix(TDP-3076/backend): Implement performance fix for studio

### PR DESCRIPTION
**Pull request by fhuaulme**
https://jira.talendforge.org/browse/TDP-3076

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed

**Please check the browsers you've tested on**
- [x] No, and no need to (backend changes only)

[Fix]
* Implements solution #2 in TDP-3076's proposed fixes (new Preparation format).
* Unit test for new format.

[Misc]
* Minor code format.
* Better time measure for callable operations (async).

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/751/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/751/)

File: dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationController.java [hunk 1]
```diff
@@ -100,6 +100,20 @@ public String create(@ApiParam("preparation") @RequestBody final Preparation pre
     }
 
     /**
+     * List all preparation details.
```
Comment (skermabon) [inline dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationController.java]:
```     * List all preparation summaries.
     *
     * @return the preparation summarie,  sorted by descending last modification date.
```

File: dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationController.java [hunk 2]
```diff
@@ -100,6 +100,20 @@ public String create(@ApiParam("preparation") @RequestBody final Preparation pre
...
+     *
+     * @param sort  how to sort the preparations.
+     * @param order how to order the sort.
+     * @return the preparation details.
+     */
+    @RequestMapping(value = "/preparations/summaries", method = GET)
+    @ApiOperation(value = "List all preparations", notes = "Returns the list of preparations details the current user is allowed to see. Creation date is always displayed in UTC time zone. This operation return all details on the preparations.")
```
Comment (skermabon) [inline dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationController.java]:
```
@ApiOperation(value = "List all preparations", notes = "Returns the list of preparations summaries the current user is allowed to see. Creation date is always displayed in UTC time zone. This operation return the summary on the preparations.")
```

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/761/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/761/)

**Pull request merged**